Make duplicate flush recalls success and other logic update

soothing-rain opened the pull request:
Latest logic:
1) Duplicate flush calls on same segment will not result in errors (same as the original design)
2) `FlushSegments` now still flushes stale segments even if some non-stale segment failed to get flushed

issue: #16749

/kind enhancement

Signed-off-by: Yuchen Gao <yuchen.gao@zilliz.com>

soothing-rain commented:
/assign @xiaofan-luan

xiaofan-luan commented on internal/datanode/data_node.go:
```diff
@@ -547,48 +547,51 @@ func (node *DataNode) FlushSegments(ctx context.Context, req *datapb.FlushSegmen
 
+			// Cache the segment and send it to its flush channel.
+			flushedSeg = append(flushedSeg, segID)
+			node.segmentCache.Cache(segID)
```
should we use segmentCache.Cache() and node.segmentCache.checkIfCached() in one call to make it atomically

soothing-rain replied:
Sure i have made it atomic.

soothing-rain committed:
Make duplicate flush recalls success and other logic update

Latest logic:
1) Duplicate flush calls on same segments will not result in errors (same as the original design)
2) `FlushSegments` now still flushes stale segments even if non-stale segments failed to get flushed

issue: #16749

/kind enhancement

Signed-off-by: Yuchen Gao <yuchen.gao@zilliz.com>

soothing-rain commented:
rerun ut

xiaofan-luan commented:
/lgtm
/approve
